app/lib/backend_api_logic/service_extension.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module BackendApiLogic
+  module ServiceExtension
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :backend_api_configs, inverse_of: :service, dependent: :destroy
+      has_many :backend_apis, through: :backend_api_configs, dependent: :destroy
+
+      alias_method :backend_api, :first_backend_api
+    end
+
+    def first_backend_api
+      @backend_api ||= find_or_create_first_backend_api!
+    end
+
+    def find_or_create_first_backend_api!
+      return unless account
+      config = backend_api_configs.first || create_first_backend_api_config!
+      config.backend_api
+    end
+
+    private
+
+    def create_first_backend_api_config!
```

Comment:
I think that this does not have a test and that it dangerous 😞 
![image](https://user-images.githubusercontent.com/11318903/62144327-e330cf00-b2f1-11e9-99f5-6142deb88fc4.png)

Reply:
wait, maybe it is tested in here:
```
test/unit/services/service_discovery/import_cluster_definitions_service_test.rb
96:      @service.backend_api.stubs(account: @account)
```
I will see 👍

Reply:
`create_first_backend_api_config!` is private, but `find_or_create_first_backend_api!` should have a test for sure!

Reply:
ok...
![image](https://user-images.githubusercontent.com/11318903/62144540-4cb0dd80-b2f2-11e9-98d6-ef834c5e73d2.png)

so this means that there is at least 1 test that relies on this somehow:
```ruby
test 'unsupported backend base path' do
      @import_service.stubs(cluster_service: @cluster_service)

      @account.stubs(:provider_can_use?).with(:apicast_v1).returns(true)
      @account.stubs(:provider_can_use?).with(:apicast_v2).returns(true)
      @account.stubs(:provider_can_use?).with(:proxy_private_base_path).returns(false)
      @service.backend_api.stubs(account: @account)

      System::ErrorReporting.expects(:report_error).with(responds_with(:message, 'Could not save API backend URL'), any_parameters)
      @import_service.send(:import_cluster_service_endpoint_to, @service)
      refute @service.proxy.valid?
      assert_equal 'http://api.example.net:80', @service.proxy.reload.api_backend
    end
```

But imo we still need one test that tests this in isolation and a bit more in depth 😄